server/util/tracing/tracing.go

```diff
@@ -309,3 +245,53 @@ func AddStringAttributeToCurrentSpan(ctx context.Context, key, value string) {
 	}
 	span.SetAttributes(attribute.String(key, value))
 }
+
+func parseOverrides() error {
+	for _, override := range *traceFractionOverrides {
+		parts := strings.Split(override, "=")
+		if len(parts) != 2 {
+			return status.InvalidArgumentErrorf("Trace fraction override %q has invalid format, expected name=fraction", override)
+		}
+		name := parts[0]
+		fraction, err := strconv.ParseFloat(parts[1], 64)
+		if err != nil {
+			return status.InvalidArgumentErrorf("Trace fraction override %q has invalid format, fraction not valid: %s", override, err)
+		}
+		overrideFractions[name] = fraction
+	}
+	return nil
+}
+
+// shouldTrace returns a boolean indicating if request should be traced because:
+//  - tracing was forced
+//  - some other service is tracing this request already
+//  - sampling indicates this request should be traced
+func ShouldTrace(ctx context.Context, method string) bool {
+	grpcMD, ok := metadata.FromIncomingContext(ctx)
```

Comment:
@vadimberezniker i'm a little unsure if this is right for the client interceptors -- should those be using FromOutgoingContext?

Reply:
Fixed.